### PR TITLE
Separate memory and kmalloc modules

### PR DESCRIPTION
**Pull request by shuyangli**
Separate memory and kmalloc modules into different files. Because the memory_page function series will be used as the basis for other functions in user land, and I'm about to write some of them, it makes sense to make the separation before the modules get too big.

Discussion: are umbrella headers useful/necessary?

### CODE REVIEW

File: src/interrupt.c [hunk 1]
```diff
@@ -39,12 +39,17 @@ static void unknown_exception(int i, int code) {
 
     if (i == 14) {
         asm("mov %%cr2, %0":"=r"(vaddr));
+        // When a page fault exception is thrown, test if the vaddr is mapped
```
Comment (kkoser) [inline src/interrupt.c]: These comments seem unrelated to the point of this PR

Comment (shuyangli) [inline src/interrupt.c]: Do you want a separate pr for comments only?

**Comment by shuyangli** [comment]
Don't merge; changes coming

File: src/memory.h [hunk 2]
```diff
@@ -7,41 +7,7 @@ See the file LICENSE for details.
...
-
-/**
-* @brief Frees previously allocated memory in the kernel's portion of memory
-* @detail Takes the given pointer, and finding the appropriate page in memory, marks that region as free on that page's kmalloc_page_info.
-*
-* @param to_free The pointer to the segment of memory to be free'd. This should have been obtained from a call to kmalloc.
-* @return void
-*/
-void kfree(void* to_free);
+#include "memory_raw.h"
```
Comment (kkoser) [inline src/memory.h]: After discussion, we vote to remove this line as memory_raw is internal and should not be exposed as part of general memory usage.

**Comment by kkoser** [comment]
LGTM! +1
